demo/android/STT/porcupine-stt-demo-app/src/main/java/ai/picovoice/porcupine/sttdemo/MainActivity.java

```diff
@@ -0,0 +1,320 @@
+    limitations under the License.
+*/
+
+package ai.picovoice.porcupine.sttdemo;
+
+
+import android.Manifest;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.os.Bundle;
+import android.os.CountDownTimer;
+import android.speech.RecognitionListener;
+import android.speech.RecognizerIntent;
+import android.speech.SpeechRecognizer;
+import android.util.Log;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.ScrollView;
+import android.widget.Spinner;
+import android.widget.TextView;
+import android.widget.Toast;
+import android.widget.ToggleButton;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.core.app.ActivityCompat;
+
+import java.util.ArrayList;
+import java.util.Locale;
+
+import ai.picovoice.porcupine.Porcupine;
+import ai.picovoice.porcupine.PorcupineException;
+import ai.picovoice.porcupine.PorcupineManager;
+import ai.picovoice.porcupine.PorcupineManagerCallback;
+
+
+public class MainActivity extends AppCompatActivity {
+    private PorcupineManager porcupineManager = null;
+
+    private ScrollView intentWrapper;
+    private TextView intentTextView;
+    private TextView stateTextView;
+
+    private SpeechRecognizer speechRecognizer;
+    private Intent speechRecognizerIntent;
+
+    private void displayError(String message) {
+        Toast.makeText(this, message, Toast.LENGTH_SHORT).show();
+    }
+
+    private PorcupineManager initPorcupine() throws PorcupineException {
+        final Spinner mySpinner = findViewById(R.id.keyword_spinner);
+        final String keywordName = mySpinner.getSelectedItem().toString();
+
+        stateTextView.setText("Listening for Wake Word");
+
+        Porcupine.BuiltInKeyword keyword = Porcupine.BuiltInKeyword.valueOf(keywordName.toUpperCase().replace(" ", "_"));
+
+        return new PorcupineManager.Builder()
+                .setKeyword(keyword)
+                .setSensitivity(0.7f)
+                .build(getApplicationContext(), new PorcupineManagerCallback() {
+                    @Override
+                    public void invoke(int keywordIndex) {
+                        runOnUiThread(new Runnable() {
+                            @Override
+                            public void run() {
+                                stateTextView.setText("Wake Word Detected");
+                                try {
+                                    // needs to stop porcupine manager before speechRecognizer can start listening.
+                                    porcupineManager.stop();
+                                } catch (PorcupineException e) {
+                                    displayError("Failed to stop porcupine.");
+                                    return;
+                                }
+
+                                new CountDownTimer(1000, 1000) {
```

Comment:
I found the delay between wake word and STT a bit long. I tried it with no delay and it felt much better. I say just remove the timer and fire off STT right away. Unless it's there for some audio delay issue.